src/zenml/models/pipeline_run_models.py

```diff
@@ -150,6 +150,25 @@ def produced_artifacts(self) -> List["ArtifactResponseModel"]:
 
         return get_artifacts_of_pipeline_run(self, only_produced=True)
 
+    def wait(self, timeout: int = 3600) -> None:
+        import time
+
+        from zenml.client import Client
+
+        start_time = time.time()
+        while True:
+            current_run_state = Client().get_pipeline_run(self.id)
+            if current_run_state.status != ExecutionStatus.RUNNING:
+                for key in current_run_state.__fields_set__:
+                    setattr(self, key, getattr(current_run_state, key))
+                return
+
+            duration = time.time() - start_time
+            if timeout > 0 and duration > timeout:
+                return
+
+            time.sleep(5)
```

Comment:
might make sense to use a backoff strategy here rather than a fixed time

Reply:
Absolutely, I'm not even sure we want this method like that on this model class. As a user I would expect that to be immutable, but what I do in this method is to update the attributes after fetching an updated copy of the model - not ideal IMO.

Maybe we want an intermediate object as the return value of running a pipeline instead:
```python
something = my_pipeline()
pipeline_run = something.wait()

# The `something.wait()` happens in the pipeline call
pipeline_run = my_pipeline(sync=True)
```

Reply:
Yeah not a big fan of the model fetching the info continuously - but is this even necessary? IMO it's fine to just write a pipeline run with something like `ExecutionStatus.PENDING` or similar to indicate that this is not actually running yet and let the user fetch the run again whenever he wants

Reply:
TL;DR I'd suggest we delete this method entirely and let the user figure it out ;)